Sources/Penny/BotStateManager.swift

```diff
@@ -26,96 +26,121 @@ import Logging
      the new instance will start handling events without waiting more for the old instance.
  */
 actor BotStateManager {
-
     let id = Int(Date().timeIntervalSince1970)
-    let services: HandlerContext.Services
+    let context: HandlerContext
     let disableDuration: Duration
     let logger: Logger
+    private var cachesPopulationContinuations: [CheckedContinuation<Void, Never>] = []
 
     var canRespond = false
-    var onStarted: (() async -> Void)?
 
     init(
-        services: HandlerContext.Services,
+        context: HandlerContext,
         disabledDuration: Duration = .seconds(3 * 60)
     ) {
-        self.services = services
+        self.context = context
         self.disableDuration = disabledDuration
         var logger = Logger(label: "BotStateManager")
         logger[metadataKey: "id"] = "\(self.id)"
         self.logger = logger
     }
 
-    func start(onStarted: @Sendable @escaping () async -> Void) async {
-        self.onStarted = onStarted
-        Task { await send(.shutdown) }
-        cancelIfCachePopulationTakesTooLong()
+    func start() {
+        switch Constants.deploymentEnvironment {
+        case .local:
+            break
+        case .prod:
+            self.context.backgroundProcessor.process {
+                await self.cancelIfCachePopulationTakesTooLong()
+            }
+            self.context.backgroundProcessor.process {
+                await self.send(.shutdown)
+            }
+        }
```

Comment:
Some env checking to make sure i can manually run Penny without worrying about interrupting the prod penny.
Other than that the file changes just move stuff to structured concurrency.